content/blog/lift-and-transform-apps-with-hpe-csi-driver-for-kubernetes.md

```diff
@@ -0,0 +1,283 @@
+INFO
+TESTING INSTALLATION
+computation_time
+00:00:49
+summary	result
+CRC	OK
+count	OK
+```
+
+If things dont’ come up properly, which is very likely it won’t on the first couple of iterations, read logs, error messages and course correct accordingly until the database come up cleanly. At this point, in a production scenario, test clients could be attached to further validate and solidify that moving the application to Kubernetes will actually work.
+
+# The endgame
+
+Up until this step, there’s been zero downtime or disruption to the source application. And the last phase is a one-way street. Ensure clients can connect to the destination database instance before proceeding.
+
+While it’s harmless to leave the test clone chart and `PersistentVolumeClaim` on the cluster, we’ll remove it to avoid confusion.
+
+```markdown
+$ helm uninstall my-test-clone
+$ kubectl delete pvc/data-my-test-clone-mariadb-0
+```
+
+Next, we need to prepare the `values-prod.yaml` file.
+
+```yaml
+image:
+  tag: 10.3
+auth:
+  rootPassword: my-password
+global:
+  imagePullSecrets: # Only needed if you're rate limited to Docker Hub
+  - regcred
+primary:
+  persistence:
+    enabled: true
+    storageClass: hpe-transform
+    annotations:
+      csi.hpe.com/importVolumeName: virt-mm-db-1.vmdk
+      csi.hpe.com/forceImport: "true"
+    size: 64Gi
+    subPath: lib
+```
+
+The only change we need to make is the `annotations` stanza. Changing the directive from `importVolAsClone` to `importVolumeName`. We also need to “force” the import as the volume we’re importing already have residual vSphere application metadata associated with it. That will be overwritten by the HPE CSI Driver when forcing the import.
+
+Before deploying the new chart, the source VM needs to be shut down and removed from vSphere control. With HPE Nimble Storage and HPE Alletra 6000 it’s perfectly safe to orderly shut down the VM and “Delete from Disk” in the vCenter context menus. If you’re attempting these workflows and procedures with HPE Alletra 9000, Primera or 3PAR. Only use “Remove from Inventory” to prevent disruption.
+
+Also, remember to have backups or volume replicas and a contingency plan for any unplanned event.
+
+Once the VM has been deleted from vSphere, the remnants on the storage array can be seen as offline volumes. The metadata volumes have been permanently removed.
+
+![Offline HPE Storage Nimble volume](/img/offline-screen-shot-2022-06-14-at-5.45.27-pm.png "Offline HPE Storage Nimble volume")
+
+An important detail to understand is that `importVolumeName` only works for offline volumes on HPE Alletra 6000 and Nimble Storage.
+
+Ok, let’s import the database into its final state.
+
+```markdown
+$ helm install my-prod bitnami/mariadb -f values-import.yaml
+```
+
+Once the database is up, we can connect to the database and verify the contents yet again.
+
+```
```

Comment:
The Markdown syntax is missing. Please add markdown or bash as syntax for the code snippet.